packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -112,176 +102,140 @@ const propTypes = {
-  constructor(props) {
-    super(props);
-
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.escapeKey = KeyCode.KEY_ESCAPE;
-  }
-
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleKeydown);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
-  }
-
-  handleKeydown(e) {
-    const notificationDialog = document.querySelector('[data-terra-notification-dialog]');
-
-    if (e.keyCode === this.escapeKey) {
-      if (notificationDialog) {
-        if (e.target === notificationDialog || notificationDialog.contains(e.target)) {
-          e.stopImmediatePropagation();
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+  useEffect(() => {
+    function handleKeydown(e) {
+      const notificationDialog = document.querySelector('[data-terra-notification-dialog]');
+
+      if (e.keyCode === KEY_ESCAPE) {
+        if (notificationDialog) {
+          if (e.target === notificationDialog || notificationDialog.contains(e.target)) {
+            e.stopImmediatePropagation();
+          }
         }
       }
     }
-  }
 
-  render() {
-    if (!this.props.isOpen) {
-      return null;
-    }
+    document.addEventListener('keydown', handleKeydown);
+
+    return (() => {
+      document.removeEventListener('keydown', handleKeydown);
+    });
+  });
+
+  const {
+    title,
+    startMessage,
+    endMessage,
+    content,
+    acceptAction,
+    rejectAction,
+    variant,
+    isOpen,
+    buttonOrder,
+    emphasizedAction,
+    custom,
+    ...customProps
+  } = props;
+
+  const signalWord = variant === variants.CUSTOM ? (custom || {}).signalWord : <FormattedMessage id={`Terra.notification.dialog.${variant}`} />;
+
+  if (process.env.NODE_ENV !== 'production' && acceptAction === undefined && rejectAction === undefined) {
+    // eslint-disable-next-line no-console
+    console.warn('The `acceptAction` and/or `rejectAction` props must be provided to render the Notification dialog');
+  }
 
-    const {
-      header,
-      title,
-      startMessage,
-      endMessage,
-      content,
-      acceptAction,
-      rejectAction,
-      variant,
-      customIcon,
-      isOpen,
-      buttonOrder,
-      emphasizedAction,
-      primaryAction,
-      secondaryAction,
-      message,
-      ...customProps
-    } = this.props;
-    if (process.env.NODE_ENV !== 'production' && acceptAction === undefined && primaryAction === undefined && rejectAction === undefined && secondaryAction === undefined) {
+  if (!isValidVariant(variant)) {
+    if (process.env.NODE_ENV !== 'production') {
       // eslint-disable-next-line no-console
-      console.warn('At least one of `acceptAction`,`primaryAction`,`rejectAction`,`secondaryAction` props must be provided for Notification dialog');
+      console.warn('The `variant` prop must be provided to render the Notification dialog');
     }
+    return null;
```

Comment:
IMO the propType validation should give devs a sufficient warning that something is wrong, and returning null here is just subtly hiding problems. If we don't want to allow users to render without a variant, we should throw an exception. That's about as subtle as a brick to the forehead.

Reply:
I updated to throw a hard error when isOpen is also true https://github.com/cerner/terra-framework/pull/1105/commits/710f82574445d978a9d549e57f97b2bdedef804e#diff-cb62dc7ce0197bc569a9952450bb9fe5R160

Reply:
removed errors based on feedback in this thread: https://github.com/cerner/terra-framework/pull/1105#discussion_r424676883